lib/pub_sub/breaker.rb

```diff
@@ -0,0 +1,47 @@
+module PubSub
+
+  class Breaker
+
+    THREAD_LOCAL_IDENTIFIER = :pub_sub_region
+
+    class << self
+
+      def current_breaker
+        all_breakers[current_breaker_idx]
+      end
+
+      def current_region
+        all_regions[current_breaker_idx]
+      end
+
+      def use_next_breaker
+        Thread.current[THREAD_LOCAL_IDENTIFIER] += 1
+        Thread.current[THREAD_LOCAL_IDENTIFIER] = Thread.current[THREAD_LOCAL_IDENTIFIER] % all_breakers.count
```

Comment:
Could also condense the above two lines into one if you wanted:

`Thread.current[THREAD_LOCAL_IDENTIFIER] = (Thread.current[THREAD_LOCAL_IDENTIFIER] +1) % all_breakers.count`